src/components/profile/Preview/index.tsx

```diff
@@ -0,0 +1,104 @@
+import { Typography } from '@/components/common';
+import { PublicProfile } from '@/lib/types/apiResponses';
+import { SocialMediaType } from '@/lib/types/enums';
+import { getLevel, getProfilePicture, isSrcAGif } from '@/lib/utils';
+import DevpostIcon from '@/public/assets/icons/devpost-icon.svg';
+import Image from 'next/image';
+import { AiOutlineLink } from 'react-icons/ai';
+import { BsFacebook, BsGithub, BsInstagram, BsLinkedin, BsTwitter } from 'react-icons/bs';
+import { IoMail } from 'react-icons/io5';
+import styles from './style.module.scss';
+
+function socialIcon(type: SocialMediaType) {
+  switch (type) {
+    case SocialMediaType.DEVPOST:
+      return <DevpostIcon className={styles.icon} aria-label="Devpost" />;
+    case SocialMediaType.EMAIL:
+      return <IoMail className={styles.icon} aria-label="Email" />;
+    case SocialMediaType.FACEBOOK:
+      return <BsFacebook className={styles.icon} aria-label="Facebook" />;
+    case SocialMediaType.GITHUB:
+      return <BsGithub className={styles.icon} aria-label="GitHub" />;
+    case SocialMediaType.INSTAGRAM:
+      return <BsInstagram className={styles.icon} aria-label="Instagram" />;
+    case SocialMediaType.LINKEDIN:
+      return <BsLinkedin className={styles.icon} aria-label="LinkedIn" />;
+    case SocialMediaType.PORTFOLIO:
+      return <AiOutlineLink className={styles.icon} aria-label="Portfolio" />;
+    case SocialMediaType.TWITTER:
+      return <BsTwitter className={styles.icon} aria-label="Twitter" />;
+    default:
+      return null;
+  }
+}
```

Comment:
We might want to move this out into a separate component since I think this'll be useful for the actual profile page